exporter/sematextexporter/writer.go

```diff
@@ -0,0 +1,241 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sematextexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sematextexporter"
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb-observability/common"
+	"github.com/influxdata/influxdb-observability/otel2influx"
+	"github.com/influxdata/line-protocol/v2/lineprotocol"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+var _ otel2influx.InfluxWriter = (*sematextHTTPWriter)(nil)
+
+type sematextHTTPWriter struct {
+	encoderPool sync.Pool
+	httpClient  *http.Client
+
+	httpClientSettings confighttp.ClientConfig
+	telemetrySettings  component.TelemetrySettings
+	writeURL           string
+	payloadMaxLines    int
+	payloadMaxBytes    int
+	hostname           string
+	token              string
+	logger             common.Logger
```

Comment:
I notice you are using an external logger, have you considered looking into logging via opentelemetry logs instead? might be more idiomatic for an OTel component to use OTel instrumentation when possible
https://github.com/open-telemetry/opentelemetry-go-instrumentation